tripy/tripy/types.py

```diff
@@ -0,0 +1,40 @@
+#
+# SPDX-FileCopyrightText: Copyright (c) 2024-2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""
+Special type annotations used in Tripy.
+"""
+
+import numbers
+import sys
+from typing import Union, Sequence
+
+from tripy import export
+
+export.public_api()(sys.modules[__name__])
+
+tensor_literal = export.public_api(
```

Comment:
Let's use PascalCase for types IMO:
```suggestion
TensorLiteral = export.public_api(
```
Also, "tensor literal" seems a bit misleading, since the type is actually just a Python collection, and *not* a tensor

Reply:
Yeah that's a good point on the name. I'm not sure what the best alternative would be. `NumberArray`, perhaps? Numpy calls such values "array-like"

Reply:
It's really just a nested sequence of numbers. Maybe `NumberSequence` or `NestedNumberSequence`?

Reply:
I'll go with `NestedNumberSequence`. A little wordy but users would probably want to use `TensorLike` regardless.